### PR TITLE
Add attack animations for Adam Blackwell

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds attack animations to the main character, Adam, for melee and ranged attacks, from all the relevant directions (up, down, left and right).

OPA-327 and OPA-328.

### CODE REVIEW

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Westlanderz) [reply Assets/Scripts/Player/UnitGridCombat.cs:169]: After that I also think it is better to call that direction selecting in the shootunit function cause that has the check if the player is actually already dead that you are trying to shoot.

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Westlanderz) [inline Assets/Scripts/Player/UnitGridCombat.cs:179]: For all the int casts here, nothing wrong about them but it might be better to do `(int)(attackerPosition.x - enemyPosition.x)`. But again I dont know for sure.

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Westlanderz) [inline Assets/Scripts/Player/UnitGridCombat.cs:345]: Not sure if that works with the animations but i think it is easier to make a function were you set all the bools to false and call it first. And after that just set the ones you want to true, the result will be a bit better to read. Basically what the `else` does below.

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Westlanderz) [reply Assets/Scripts/Player/UnitGridCombat.cs:179]: Or like the rounding i have done in the sortingorder on line [23](https://github.com/Online-Pluisje-Art/Operation-Blackwell/blob/development/Assets/Scripts/Core/Utils/PositionSortingOrder.cs#L23).

**Review by Westlanderz — CHANGES_REQUESTED** [review]
In general the code looks okay, just made some suggestions to look into.

**Review by drfdaniel — DISMISSED** [review]
LGTM

File: Assets/Scenes/TutorialLevel.unity
Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity:5427]: The sound controller was disabled, and I re-enabled it again. This was committed by accident in 74911ca

File: Assets/Scenes/TutorialLevel.unity
Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity:9560]: This disables the second Adam added in 19784d0

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Dennisbonke) [reply Assets/Scripts/Player/UnitGridCombat.cs:345]: I see what you mean yeah, that should work so I'll do that.

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Dennisbonke) [reply Assets/Scripts/Player/UnitGridCombat.cs:179]: Works too, I'll change it.

File: Assets/Scenes/TutorialLevel.unity
Comment (Westlanderz) [reply Assets/Scenes/TutorialLevel.unity:9560]: Why even disable the second Adam, that character is used to show the working of the initiative system.

Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity:9560]: Because we only have one main player in this level and it looks weird to see two of them.

Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity:9560]: If we want to show this on the review, enable this in a specialized build, but imo don't commit it.

Comment (Westlanderz) [reply Assets/Scenes/TutorialLevel.unity:9560]: I get that but what if we would like to show the initiative system work in demo, how will you demonstrate different orders in the players if you only have one player. Once that is shown we can delete the 2nd one imo.

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Dennisbonke) [reply Assets/Scripts/Player/UnitGridCombat.cs:169]: Yeah makes sense, will do.

File: Assets/Scenes/TutorialLevel.unity
Comment (Westlanderz) [reply Assets/Scenes/TutorialLevel.unity:9560]: As it is only a alpha build of the game anyway, it does not matter if it contains "debug" objects like a second Adam.

Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity:9560]: While it might be an alpha build, stuff that one only adds for a local demo shouldn't end up in github unless it is meant to be. I guess we can leave it enabled for now considering the review is Monday, but for the next time I'd rather not see these kinds of debug stuff in here. Is it fine if I remove the second Adam / disable it in the commit that starts the 0.5 development cycle?

Comment (Westlanderz) [reply Assets/Scenes/TutorialLevel.unity]: Yes but remove it not disable, as that avoids clutter in the scene file

Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity]: Got it, I’ve re-enabled it for now.

**Review by Westlanderz — CHANGES_REQUESTED** [review]
Just a few things that need answers. But I do not see any code style issues.

File: Assets/Scenes/TutorialLevel.unity
Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity:4233]: I believe we came to the conclusion that this won't have any benefit? If I remember that wrong, please let me know.

Comment (Westlanderz) [reply Assets/Scenes/TutorialLevel.unity:4233]: There was no conclusion, so I think it is best to set it in the prefab directly.

Comment (Dennisbonke) [reply Assets/Scenes/TutorialLevel.unity]: Got it, fixed that now.

File: Assets/Scripts/Player/UnitGridCombat.cs
Comment (Dennisbonke) [reply Assets/Scripts/Player/UnitGridCombat.cs:73]: You're right, fixed this.

**Review by Westlanderz — APPROVED** [review]
LGTM

**Comment by Dennisbonke** [comment]
Do not merge until #76 and #77 are merged.

**Comment by Dennisbonke** [comment]
Or nevermind

**Pull request merged**
